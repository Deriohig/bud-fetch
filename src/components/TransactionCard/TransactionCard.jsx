import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

const TransactionWrapper = styled.div`
    margin-bottom: 16px;
    width: 240px;
    background-color: #e0e0e0;
    padding: 16px;
    margin-right: 16px;
`;

const TransactionItem = styled.span`
    margin-right: 16px;
`;

const TransactionCategory = styled.p`
    font-size: 12px;
`;

const TransactionHeader = styled.div`
    display: flex;
    justify-content: space-between;
    margin-bottom: 20px;
`;

const Cost = styled.p`
    margin-top: 8px;
`;

const TransactionCard = ({ date, currency, value, description, category }) => {
    const renderCost = (value, currency) => `${value} ${currency}`
    return (
        <TransactionWrapper data-testid="TransactionCard">
            <TransactionHeader>
                <TransactionCategory>{category}</TransactionCategory>
                <TransactionCategory>{date}</TransactionCategory>
            </TransactionHeader>
            <TransactionItem>{description}</TransactionItem>
            <Cost>{renderCost(value, currency)}</Cost>
        </TransactionWrapper>
    );
};

TransactionCard.propTypes = {
    date: PropTypes.string.isRequired, 
    currency: PropTypes.string.isRequired, 
    value: PropTypes.number.isRequired, 
    description: PropTypes.string.isRequired, 
    category: PropTypes.string.isRequired
}

export default TransactionCard;
