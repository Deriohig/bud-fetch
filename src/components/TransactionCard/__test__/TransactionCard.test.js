import React from 'react';
import ReactDOM from 'react-dom';
import TransactionCard from '../index';
import {render, cleanup} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer'

const props = {
    date: '18/02/1993',
    currency: 'USD',
    value: '81.43',
    description: 'Chocolate',
    category: 'Food'
}

afterEach(cleanup);

it("renders without crashing", ()=> {
    const div = document.createElement('div');
    ReactDOM.render(<TransactionCard {...props} />, div)
})

it("renders a date", ()=> {
    const { getByTestId } = render(<TransactionCard {...props} />)
    expect(getByTestId('TransactionCard')).toHaveTextContent('18/02/1993');
})

it("renders a currency", ()=> {
    const { getByTestId } = render(<TransactionCard {...props} />)
    expect(getByTestId('TransactionCard')).toHaveTextContent('USD');
})

it("renders a value", ()=> {
    const { getByTestId } = render(<TransactionCard {...props} />)
    expect(getByTestId('TransactionCard')).toHaveTextContent('81.43');
})

it("renders a category", ()=> {
    const { getByTestId } = render(<TransactionCard {...props} />)
    expect(getByTestId('TransactionCard')).toHaveTextContent('Food');
})

it("renders a description", ()=> {
    const { getByTestId } = render(<TransactionCard {...props} />)
    expect(getByTestId('TransactionCard')).toHaveTextContent('Chocolate');
})

it("matches snapshot", ()=> {
    const snapshot = renderer.create(<TransactionCard {...props} />).toJSON();
    expect(snapshot).toMatchSnapshot();
})
