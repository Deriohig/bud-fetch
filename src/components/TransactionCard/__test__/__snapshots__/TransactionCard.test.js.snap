// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="sc-bdfBwQ gppxLM"
  data-testid="TransactionCard"
>
  <div
    className="sc-hKgILt fscPhC"
  >
    <p
      className="sc-dlfnbm kOsDjC"
    >
      Food
    </p>
    <p
      className="sc-dlfnbm kOsDjC"
    >
      18/02/1993
    </p>
  </div>
  <span
    className="sc-gsTCUz jfeqWp"
  >
    Chocolate
  </span>
  <p
    className="sc-eCssSg jgkwoT"
  >
    81.43 USD
  </p>
</div>
`;
